### DISCUSSION TITLE
How does clicks on toolbar are handled?

### DISCUSSION THREAD

**Discussion by luca-venturini** [xournalpp/xournalpp#4328] [General]
I cannot understand how clicks on buttons on the toolbar are handled, for example if I click on "Grid Snapping", and then I move between "Text" and "Pen", it remains selected. How could I obtain the same result if I want to add a button in which my functionality must remain selected regardless of the clicks on other buttons?

For now if I click my button and then I click another one, like "Text", my button appears as unselected.

Thanks

**Comment by rolandlo** [comment]
Are you talking about a new button that you created for Xournal++? Just make sure it has its own group. The "Text" button belongs to the `GROUP_TOOL` like many others. The "Grid snapping" button belongs to the `GROUP_GRID_SNAPPING`, which other buttons do not use. 

As for how clicks on toolbar buttons are handled please see `AbstractToolItem::selected` which uses `gtk_toggle_tool_button_set_active` under the hood.
